static/js/graphs_page.js

```diff
@@ -437,10 +438,82 @@ var graphPage = {
          */
         graphPage.cyGraph = graphPage.contructCytoscapeGraph();
 
+        if('legend' in style_json){
+            graphPage.cyLegend = graphPage.constructLegend(style_json);
+        }
+
         graphPage.cyGraph.panzoom();
 
         utils.initializeTabs();
 
+        var update_legend_format = 0;
+        $("#graphDetailsTabBtn").click(function () {
+            if(graph_json['data']['description']) {
+
+                var html_legend_data = graph_json['data']['description'];
+                html_legend_data = $($.parseHTML(html_legend_data));
+
+                try {
```

Comment:
Refactor the logic to small function. Move the logic to a module specific to legend.